### PR TITLE
testutils: remove gocheck

### PR DESCRIPTION
**Pull request by lmb**
Use testing instead of gopkg.in/check.v1. Migrate users of IntegrationCheck to IntegrationTest.

### CODE REVIEW

**Comment by lmb** [comment]
ci-l4lb is https://github.com/cilium/cilium/issues/25892

File: pkg/testutils/condition_test.go
Comment (giorio94) [inline pkg/testutils/condition_test.go:1]: Nit. Should these changes be in a separate commit?

**Review by giorio94 — APPROVED** [review]
Thanks!

**Review by nebril — APPROVED** [review]

**Comment by lmb** [comment]
@tommyp1ckles could you give this PR a quick one over please?

File: pkg/testutils/condition_test.go
Comment (lmb) [reply pkg/testutils/condition_test.go:1]: You mean, split removing checkmate from testutils from removing `IntegrationCheck`?

Comment (giorio94) [reply pkg/testutils/condition_test.go:1]: Yep, exactly (if that makes sense to you)

**Comment by lmb** [comment]
ci-ginkgo is https://github.com/cilium/cilium/issues/26009

**Review by ti-mo — APPROVED** [review]
:100: :100: :100:

**Review by sayboras — APPROVED** [review]
LGTM :heavy_check_mark:

**Comment by lmb** [comment]
/test-1.26-net-next

**Review by aanm — APPROVED** [review]

**Review by tommyp1ckles — APPROVED** [review]

**Comment by tommyp1ckles** [comment]
sorry for the delay - looks good
